clang/test/SemaSYCL/intel-fpga-local-ast.cpp

```diff
@@ -0,0 +1,537 @@
+  //CHECK: VarDecl{{.*}}max_replicates
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelMaxReplicatesAttr
+  //CHECK: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}2
+  //CHECK: IntegerLiteral{{.*}}2{{$}}
+  [[intel::max_replicates(2)]] unsigned int max_replicates[64];
+
+  //CHECK: VarDecl{{.*}}dual_port
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelSimpleDualPortAttr
+  [[intel::simple_dual_port]] unsigned int dual_port[64];
+
+  //CHECK: VarDecl{{.*}}arr_force_p2d_0
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelForcePow2DepthAttr
+  //CHECK: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}0
+  //CHECK: IntegerLiteral{{.*}}0{{$}}
+  [[intel::force_pow2_depth(0)]] unsigned int arr_force_p2d_0[64];
+
+  //CHECK: VarDecl{{.*}}arr_force_p2d_1
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelForcePow2DepthAttr
+  //CHECK: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}1
+  //CHECK: IntegerLiteral{{.*}}1{{$}}
+  [[intel::force_pow2_depth(1)]] unsigned int arr_force_p2d_1[64];
+  [[intel::fpga_register]] int var_reg;
+  [[intel::numbanks(4), intel::bankwidth(16), intel::singlepump]] int var_singlepump;
+  [[intel::numbanks(4), intel::bankwidth(16), intel::doublepump]] int var_doublepump;
+  [[intel::numbanks(4), intel::bankwidth(16)]] int var_numbanks_bankwidth;
+  [[intel::bank_bits(2, 3), intel::bankwidth(16)]] int var_bank_bits_width;
+  [[intel::max_replicates(2)]] int var_max_repl;
+  [[intel::simple_dual_port]] int var_dual_port;
+  [[intel::force_pow2_depth(1)]] int var_force_p2d;
+  [[intel::force_pow2_depth(1)]] const int const_force_p2d[64] = {0, 1};
+
+  // Check duplicate argument values with implicit memory attribute.
+  //CHECK: VarDecl{{.*}}var_max_replicates
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelMaxReplicatesAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}12
+  //CHECK-NEXT: IntegerLiteral{{.*}}12{{$}}
+  [[intel::max_replicates(12)]]
+  [[intel::max_replicates(12)]] int var_max_replicates; // OK
+
+  // Check duplicate argument values.
+  //CHECK: VarDecl{{.*}}var_private_copies
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelPrivateCopiesAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}12
+  //CHECK-NEXT: IntegerLiteral{{.*}}12{{$}}
+  [[intel::private_copies(12)]]
+  [[intel::private_copies(12)]] int var_private_copies; // OK
+
+  // Checking of duplicate argument values.
+  //CHECK: VarDecl{{.*}}var_forcep2d
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelForcePow2DepthAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}1
+  //CHECK-NEXT: IntegerLiteral{{.*}}1{{$}}
+  [[intel::force_pow2_depth(1)]]
+  [[intel::force_pow2_depth(1)]] int var_forcep2d; // OK
+
+  // Test for Intel FPGA bankwidth memory attribute duplication.
+  // No diagnostic is emitted because the arguments match.
+  // Duplicate attribute is silently ignored.
```

Comment:
If we want to check that duplicate attribute is ignored, we need a `CHECK-NOT`

Reply:
Done